### PR TITLE
Add ajax sorting

### PR DESCRIPTION
**Pull request by volkov97**

### CODE REVIEW

File: server/controllers/gh.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ function createIssue(req, res) {
         .catch(err => onError(req, res, err));
 }
 
-function getIssues(req, res) {
+function getIndex(req, res) {
     logger.log('getIssues');
```
Comment (tadatuta) [inline server/controllers/gh.js]: fixme ;)

File: common.blocks/sidebar/__sorting/sidebar__sorting.js [hunk 2]
```diff
@@ -13,16 +13,17 @@ modules.define(
...
-            if (sortType.indexOf('-reverse') !== -1) {
-                sortType = sortType.split('-')[0];
+            if (sort.indexOf('-reverse') !== -1) {
+                sort = sort.split('-')[0];
                 direction = 'asc';
             }
 
-            Location.change({ params: this._getParams(sortType, direction) });
-            window.location.reload();
+            Location.change({ params: this._getParams(sort, direction) });
```
Comment (tadatuta) [inline common.blocks/sidebar/__sorting/sidebar__sorting.js]: it should be possible to subscribe to location change and get rid of additional `sorting` event

Comment (skad0) [reply common.blocks/sidebar/__sorting/sidebar__sorting.js]: @tadatuta can you provide arguments why subscribing to location change is better?

Comment (tadatuta) [reply common.blocks/sidebar/__sorting/sidebar__sorting.js]: 'coz one source of truth is better

File: common.blocks/sidebar/sidebar.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+modules.define('sidebar', ['i-bem-dom', 'sidebar__sorting'], function(provide, bemDom, Sorting) {
```
Comment (tadatuta) [inline common.blocks/sidebar/sidebar.js]: sidebar should not be responsible for sorting

File: common.blocks/sidebar/__sorting/sidebar__sorting.deps.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-    mustDeps: ['i-bem-dom'],
-    shouldDeps: [
-        {
-            block: 'select',
-            mods: { mode: 'radio', theme: 'islands', size: 'm' }
-        },
-        'location'
-    ]
-});
+([
```
Comment (tadatuta) [inline common.blocks/sidebar/__sorting/sidebar__sorting.deps.js]: useless diff

**Pull request merged**